### PR TITLE
 ⭐️New: Add `vue/no-restricted-syntax` rule

### LINKED ISSUE #689: no-restricted-syntax linting for templates
**Tell us about your environment**	
* **ESLint version:** 4.19.1
* **eslint-plugin-vue version:** 4.4.0
* **Node version:**	8.9.4

**Please show your full configuration:**

```json
{
    "rules": {
    "no-restricted-syntax": [
      "error",
      {
        "selector": "CallExpression[callee.property.name='$gettext'] TemplateLiteral"
      }
    ]
}
```

**What did you do?**	
Apologies if this is more of an eslint than a eslint-plugin-vue issue. I am trying to write a linting rule to ensure that no template literals are passed into our gettext function (registered on Vue as vm.$gettext). This is working fine for script code, but I have played around with a number of rules and can't seem to get it to work for template invocations.

```vue
<template>
    <div :foo="$gettext(`bar`)">{{$gettext(`bar`)}}</div>
</template>

<script>
export default {
    computed: {
        baz() {
            return this.$gettext(`bar`);
        }
    }
}
</script>
```

**What did you expect to happen?**
3 linting errors, for each incorrect invocation of $gettext

**What actually happened?**
1 linting error from the script, none from the template

### PR DESCRIPTION
**Pull request by ota-meshi**
Close #689

### CODE REVIEW

File: docs/rules/no-restricted-syntax.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+This rule is the same rule as core [no-restricted-syntax] rule but it applies to the expressions in `<template>`.
+
+
+## :wrench: Options
+
+Please see [no-restricted-syntax] for detailed options.
+
+You can include the AST created by [vue-eslint-parser] in the selector.
+To know more about certain nodes in produced AST, please go [vue-eslint-parser - AST docs].
```
Comment (michalsnik) [inline docs/rules/no-restricted-syntax.md]: Can you put link to the AST docs here?

File: docs/rules/no-restricted-syntax.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+### `"VElement > VExpressionContainer CallExpression"`
+
+Forbind call expressions on mustache interpolation.
```
Comment (michalsnik) [inline docs/rules/no-restricted-syntax.md]: `Forbids call expressions inside mustache interpolation.`

File: tests/lib/rules/no-restricted-syntax.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+          message: 'Call expressions are not allowed.',
+          line: 3,
+          column: 26,
+          endLine: 3,
+          endColumn: 33
+        }
+      ]
+    },
+
+    // Forbind call expressions on mustache interpolation
```
Comment (michalsnik) [inline tests/lib/rules/no-restricted-syntax.js]: Same as above ☝️

File: docs/rules/no-restricted-syntax.md (hunk 1 above)
Comment (ota-meshi) [reply docs/rules/no-restricted-syntax.md]: It is defined in line 54.

**Comment by ota-meshi** [comment]
@michalsnik Thank you for your review.
I fixed it. so please check again.

**Pull request merged**